ocaml/xapi/cluster_stack_constraints.ml

```diff
@@ -40,8 +40,17 @@ let required_cluster_stack ~__context =
         failwith "SR type not found in SM table."
       end
     ) pbds in
+  let failwith_cluster_stack_conflict () =
+      error "Conflicting cluster stack demands.";
+      failwith "Conflicting cluster stack demands."
+  in
   match required_stacks with
-  | [] -> active_cluster_stack  (* None of the attached SRs have constraints *)
+  | [] ->
+    (* None of the attached SRs have constraints *)
+    begin match active_cluster_stack with
+    | None -> None
```

Comment:
you could just match the empty list here, not sure if there is value in transforming the empty list to None, and other lists to Some ...